pkg/server/plugin/noderesolver/aws/iid.go

```diff
@@ -49,54 +48,42 @@ type awsClient interface {
 	GetInstanceProfileWithContext(aws.Context, *iam.GetInstanceProfileInput, ...request.Option) (*iam.GetInstanceProfileOutput, error)
 }
 
-type IIDResolverConfig struct {
-	AccessKeyID     string `hcl:"access_key_id"`
-	SecretAccessKey string `hcl:"secret_access_key"`
-}
-
+// IIDResolverPlugin implements node resolution for agents running in aws.
 type IIDResolverPlugin struct {
 	mu      sync.RWMutex
-	config  *IIDResolverConfig
+	config  *caws.IIDAgentConfig
 	clients map[string]awsClient
 
 	hooks struct {
 		getenv    func(string) string
-		newClient func(config *IIDResolverConfig, region string) (awsClient, error)
+		newClient func(config *caws.IIDAgentConfig, region string) (awsClient, error)
 	}
 }
 
 var _ noderesolver.Plugin = (*IIDResolverPlugin)(nil)
 
+// NewIIDResolverPlugin creates a new IIDResolverPlugin.
 func NewIIDResolverPlugin() *IIDResolverPlugin {
 	p := &IIDResolverPlugin{}
 	p.hooks.getenv = os.Getenv
 	p.hooks.newClient = newAWSClient
 	return p
 }
 
+// Configure configures the IIDResolverPlugin
 func (p *IIDResolverPlugin) Configure(ctx context.Context, req *spi.ConfigureRequest) (*spi.ConfigureResponse, error) {
 	// Parse HCL config payload into config struct
-	config := new(IIDResolverConfig)
+	config := new(caws.IIDAgentConfig)
 	if err := hcl.Decode(config, req.Configuration); err != nil {
 		return nil, iidError.New("unable to decode configuration: %v", err)
 	}
 
 	// Set defaults from the environment
 	if config.AccessKeyID == "" {
-		config.AccessKeyID = p.hooks.getenv("AWS_ACCESS_KEY_ID")
+		config.AccessKeyID = p.hooks.getenv(caws.AccessKeyIDVarName)
 	}
 	if config.SecretAccessKey == "" {
-		config.SecretAccessKey = p.hooks.getenv("AWS_SECRET_ACCESS_KEY")
-	}
-
-	switch {
```

Comment:
why are we removing these validity checks?

Reply:
see #683, in a desire for consistency between these two plugins in regards to AWS sessions

Reply:
Ah yes, the attestor was more relaxed. I guess my question is, does it make sense to have an access key id without a secret access key or vice-versa? If it doesn't, we may consider keeping this switch and changing it to allow the case where both are empty but still failing if one is set but not the other. Thoughts?

Reply:
for posterity: https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html#access-keys-and-secret-access-keys

you are correct, having one without the other is just as useful/useless as having neither.

however, does it make sense to actually fail out on having one but not the other? I could understand making a WARN or INFO log about it, but it is not an actual failure condition since we could default to the region config only

Reply:
IMO, failing early is the right thing to do since this would most certainly be a case of misconfiguration (principal of least surprise). 

Side note: plugins do not (yet) have access to the SPIRE logging facilities.

Reply:
do you think the Attestor plugin should be altered then to also follow that logic? This would make the plugins consistent with each other but alter Attestor behavior (for a good cause)

Reply:
There is some risk to break some deployment if we do. I think the risk is small though since anybody with a working deployment with such a configuration would have already been relying on the credentials file and therefore not likely to have ever needed to specify the access key or secret access key.

I'm ok with tightening up the attestor with the same access check as the resolver.